src/client/datascience/editor-integration/hoverProvider.ts

```diff
@@ -94,13 +101,19 @@ export class HoverProvider implements INotebookExecutionLogger, vscode.HoverProv
                     const notebooks = this.getMatchingNotebooks(document);
                     if (notebooks && notebooks.length) {
                         // Just use the first one to reply if more than one.
-                        const match = await Promise.race(
-                            notebooks.map((n) => this.variableProvider.getMatchingVariable(word, n, t))
+                        const attributes = await Promise.race(
+                            notebooks.map((n) => this.variableProvider.getVariableProperties!(word, n, t))
                         );
-                        if (match) {
-                            return {
-                                contents: [`${word} = ${match.value}`]
+                        if (attributes) {
+                            const asMarkdown =
+                                Object.entries(attributes).reduce(
```

Comment:
I always forget about reduce. That's a nice way to put the string together. If it doesn't have any matching attributes does it create an empty line? Just not sure if the \n goes on the end of the entry or on the start. From the docs it looks like maybe the initial value should just be ``` and the \n would be on the start of each entry, not the end.

Reply:
If it doesn't have any matching attributes we probably shouldn't return a hover result at all.